kafka/adminapi_test.go

```diff
@@ -58,22 +65,32 @@ func TestAdminAPIWithDefaultValue(t *testing.T) {
 				ReplicationFactor: -1,
 			},
 		})
-	if err != nil {
-		adminClient.Close()
-		t.Fatalf("Failed to create topics %v\n", err)
+	if err != errCheck {
+		t.Errorf("CreateTopics() failed to return expected result")
+		if !a.isClosed {
+			a.Close()
+			t.Fatalf("Failed to create topics %v\n", err)
+		}
+	}
+	if !a.isClosed {
```

Comment:
Why are you checking isClosed here and below?

Also, these tests should preferably only use the public APIs and isClosed is not.

Reply:
We are performing same tests on closed admin client and open admin client. In case of closed client we don't need to close the client but for other case we were closing the client and producing fatal error. So to differentiate a.isClosed helps.

Reply:
@milindl need a feedback here.

Reply:
I can't see a way around checking it, if we should only use the public APIs then we can add an `IsClosed() bool` for that.